### PR TITLE
Basic SPARQL-star eval tests

### PR DESCRIPTION
**Pull request by afs**
SPARQL-star eval tests.

This is a snapshot of work-in-progress.

### CODE REVIEW

**Comment by gkellogg** [comment]
We should add "eval/manifest.jsonld" to "sparql/manifest.jsonld" and decide if "manifest-query.jsonld" and "manifest-update.jsonld" still belong.

**Comment by afs** [comment]
@gkellogg Yes (already done locally).

I'm only producing one manifests at the moment because it keeps changing; I'd like some advice on making JSON-LD for eval tests look nice when the suite is moderately stable.

**Comment by gkellogg** [comment]
There’s a make task to do it, but it just uses a standard JSON serializer. Otherwise, you could write a customized one to control white space, but I think trying to maintain that may be more trouble than it’s worth.

**Review by gkellogg — APPROVED** [review]

**Comment by afs** [comment]
This includes conversion "RDF*" to "RDF-star" in the Turtle and SPARQL test areas,  and a general test cleanup, as separate commits.

**Comment by afs** [comment]
This is ready for implementers to try out:

It covers:
* result sets
* single triple pattern matches
* basic graph patterns
* CONSTRUCT
* GRAPH

Still more to do:
* [ ] Functions
* [ ] SPARQL Update
* [ ] Fix HTML generation for eval tests
* [ ] Acceptable JSON-LD

**Review by gkellogg — CHANGES_REQUESTED** [review]
We need a sparql/eval/manifest.jsonld (and .html?). And sparql/manifest.jsonld should probably reference that version.

There's a Make task that will create both.

**Comment by afs** [comment]
> We need a sparql/eval/manifest.jsonld 

Yes - this is on the todo list above "Acceptable JSON-LD". 

Previous I have crafted a template that makes JSON that uses  "tests/manifest-context.jsonld" so is more "JSON" and less "RDF". Could make-jsonld-manifests use that? Currently, it will overwrite the existing files. 

I just haven't figured out how to make it work with the SPARQL eval requirements.

> (and .html?)

make-html-manifests.py does not work - my guess is that it is because the actions are bnodes of a pair or more triples.

**Comment by pchampin** [comment]
> make-html-manifests.py does not work - my guess is that it is because the actions are bnodes of a pair or more triples.

that's correct. I need to fix it...

**Comment by afs** [comment]
I just haven't had time so it was either delay and remain in draft or put it for review now. The status is written in the comment above.

It going to cover several areas. I thought it was more useful to have this first part now.

File: tests/sparql/eval/data-7.ttl [hunk 1]
```diff
@@ -0,0 +1,18 @@
+PREFIX : <http://example/>
+
+# One each of each collection class.
```
Comment (TallTed) [inline tests/sparql/eval/data-7.ttl:3]:
```suggestion
# One of each collection class.
```

File: tests/sparql/eval/manifest.html [hunk 2]
```diff
@@ -1448,6 +1449,72 @@ <h2>SPARQL-star - Embedded triple - Functions <a href="#sparql-star-expr-2">🔗
...
+
+SELECT * {
+   :item :p ?v .
+} ORDER BY ?v
+</pre>
+<div>Data: <code><a href="data-7.ttl">data-7.ttl</a></code></div>
+<pre>
+PREFIX : &lt;http://example/&gt;
+
+# One each of each collection class.
```
Comment (TallTed) [inline tests/sparql/eval/manifest.html:1470]:
```suggestion
# One of each collection class.
```

File: tests/sparql/eval/data-7.ttl (hunk 1 above)
Comment (afs) [reply tests/sparql/eval/data-7.ttl:3]: Doesn't read well but it is as meant.  It's not "one of the classes", its one instance of each class.

"One item of each of the collation classes." is clearer?

Comment (TallTed) [reply tests/sparql/eval/data-7.ttl:3]: My sentence does not say "one of the classes"; it effectively says "one [item] from each collection class."  (The word "item" is unnecessary, but can be included.)

"We have apples, oranges, and bananas. Which would you like?"
"One of each [collection class]."
Means you get one apple, one orange, and one banana ... which I believe is your intent.

Comment (afs) [reply tests/sparql/eval/data-7.ttl:3]: This is test data.
Changed to:
```
# The different kinds of orderable items.
# (Not in collation order).
```

**Review by gkellogg — APPROVED** [review]

**Comment by gkellogg** [comment]
Looks good, they're all passing for me now.

We need to make sure the grammar is updated to add EmbTP (or whatever it's called now) to PrimaryExpression.

**Comment by hartig** [comment]
> We need to make sure the grammar is updated to add EmbTP (or whatever it's called now) to PrimaryExpression.

It's on my agenda to work on the PR for that.

**Comment by afs** [comment]
Rebased to current main.

**Pull request merged**